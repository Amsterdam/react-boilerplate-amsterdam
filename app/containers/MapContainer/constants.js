/*
 *
 * MapContainer constants
 *
 */

export const GET_GEO = 'app/MapContainer/GET_GEO';
export const SET_GEO = 'app/MapContainer/SET_GEO';
