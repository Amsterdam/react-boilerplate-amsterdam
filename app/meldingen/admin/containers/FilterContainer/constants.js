/*
 *
 * FilterContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/FilterContainer/DEFAULT_ACTION';
