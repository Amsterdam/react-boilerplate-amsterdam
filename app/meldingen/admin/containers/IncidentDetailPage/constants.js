export const REQUEST_INCIDENT = 'app/OverviewPage/REQUEST_INCIDENT';
export const REQUEST_INCIDENT_SUCCESS = 'app/OverviewPage/REQUEST_INCIDENT_SUCCESS';
export const REQUEST_INCIDENT_ERROR = 'app/OverviewPage/REQUEST_INCIDENT_ERROR';
