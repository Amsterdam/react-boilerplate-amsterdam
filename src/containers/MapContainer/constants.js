/*
 *
 * MapContainer constants
 *
 */

export const GET_GEO = 'src/MapContainer/GET_GEO';
export const SET_GEO = 'src/MapContainer/SET_GEO';
