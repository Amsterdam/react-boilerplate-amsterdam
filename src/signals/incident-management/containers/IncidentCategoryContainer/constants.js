export const REQUEST_CATEGORY_UPDATE = 'src/IncidentStatusContainer/REQUEST_CATEGORY_UPDATE';
export const REQUEST_CATEGORY_UPDATE_SUCCESS = 'src/IncidentStatusContainer/REQUEST_CATEGORY_UPDATE_SUCCESS';
export const REQUEST_CATEGORY_UPDATE_ERROR = 'src/IncidentStatusContainer/REQUEST_CATEGORY_UPDATE_ERROR';
