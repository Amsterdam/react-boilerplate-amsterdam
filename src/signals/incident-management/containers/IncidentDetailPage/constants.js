export const REQUEST_INCIDENT = 'src/OverviewPage/REQUEST_INCIDENT';
export const REQUEST_INCIDENT_SUCCESS = 'src/OverviewPage/REQUEST_INCIDENT_SUCCESS';
export const REQUEST_INCIDENT_ERROR = 'src/OverviewPage/REQUEST_INCIDENT_ERROR';
