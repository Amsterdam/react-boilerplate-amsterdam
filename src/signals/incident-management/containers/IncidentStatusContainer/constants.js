export const REQUEST_STATUS_LIST = 'src/IncidentStatusContainer/REQUEST_STATUS_LIST';
export const REQUEST_STATUS_LIST_SUCCESS = 'src/IncidentStatusContainer/REQUEST_STATUS_LIST_SUCCESS';
export const REQUEST_STATUS_LIST_ERROR = 'src/IncidentStatusContainer/REQUEST_STATUS_LIST_ERROR';

export const REQUEST_STATUS_CREATE = 'src/IncidentStatusContainer/REQUEST_STATUS_CREATE';
export const REQUEST_STATUS_CREATE_SUCCESS = 'src/IncidentStatusContainer/REQUEST_STATUS_CREATE_SUCCESS';
export const REQUEST_STATUS_CREATE_ERROR = 'src/IncidentStatusContainer/REQUEST_STATUS_CREATE_ERROR';

