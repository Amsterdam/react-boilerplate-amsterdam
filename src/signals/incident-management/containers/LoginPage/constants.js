/*
 *
 * LoginPage constants
 *
 */

export const DEFAULT_ACTION = 'src/LoginPage/DEFAULT_ACTION';
