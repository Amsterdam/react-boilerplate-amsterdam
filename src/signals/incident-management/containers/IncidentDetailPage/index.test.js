import React from 'react';
import { shallow } from 'enzyme';

import { IncidentDetailPage, mapDispatchToProps } from './index';
import { REQUEST_INCIDENT } from './constants';
import stadsdeelList from '../../definitions/stadsdeelList';
import PrintLayout from './components/PrintLayout';


describe('<IncidentDetailPage />', () => {
  let props;

  beforeEach(() => {
    props = {
      id: '100',
      incidentdetailpage: { incident: {}, stadsdeelList },
      onRequestIncident: jest.fn()
    };
  });

  it('should render correctly', () => {
    const renderedComponent = shallow(
      <IncidentDetailPage {...props} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should change the state', () => {
    const renderedComponent = shallow(
      <IncidentDetailPage {...props} />
    );
    renderedComponent.instance().onPrintView();
    renderedComponent.instance().onTabChanged(1);
    expect(renderedComponent.instance().state).toEqual({ selectedTab: 1, printView: true });
  });

  it('should render the print view', () => {
    const renderedComponent = shallow(
      <IncidentDetailPage {...props} />
    );
    renderedComponent.instance().onPrintView();
    renderedComponent.update();
    expect(renderedComponent.find(PrintLayout).length).toEqual(1);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    // For the `mapDispatchToProps`, call it directly but pass in
    // a mock function and check the arguments passed in are as expected
    mapDispatchToProps(dispatch).onRequestIncident({});
    expect(dispatch.mock.calls[0][0]).toEqual({ type: REQUEST_INCIDENT, payload: {} });
  });
});
