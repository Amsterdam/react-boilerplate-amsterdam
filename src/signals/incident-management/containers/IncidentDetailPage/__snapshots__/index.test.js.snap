// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentDetailPage /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IncidentDetailPage
    incidentdetailpage={
      Object {
        "incident": Object {},
      }
    }
    onRequestIncident={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-12"
        >
          <h3>
            Melding 
          </h3>
        </div>,
        <ul
          className="col-4 incident-detail-page__map"
        >
          <MapDetail
            label=""
          />
        </ul>,
        <div
          className="col-8"
        >
          (
          <Link
            replace={false}
            to="undefined/incidents"
          >
            Terug naar overzicht
          </Link>
          )
          <IncidentDetail
            incident={Object {}}
          />
        </div>,
        <div
          className="col-12"
        >
          <Tabs
            onTabChanged={[Function]}
            selectedTab={0}
            tabs={
              Array [
                "Status",
                "Categorie",
              ]
            }
          />
        </div>,
        <div
          className="col-12"
        >
          <div
            className="incident-detail-page__tab-container"
          >
            <withReducer(withSaga(Connect(IncidentStatusContainer))) />
          </div>
        </div>,
      ],
      "className": "incident-detail-page row container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            Melding 
          </h3>,
          "className": "col-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Melding ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Melding ",
            undefined,
          ],
          "type": "h3",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MapDetail
            label=""
          />,
          "className": "col-4 incident-detail-page__map",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "(",
            <Link
              replace={false}
              to="undefined/incidents"
            >
              Terug naar overzicht
            </Link>,
            ")",
            <IncidentDetail
              incident={Object {}}
            />,
          ],
          "className": "col-8",
        },
        "ref": null,
        "rendered": Array [
          "(",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Terug naar overzicht",
              "replace": false,
              "to": "undefined/incidents",
            },
            "ref": null,
            "rendered": "Terug naar overzicht",
            "type": [Function],
          },
          ")",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "incident": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Tabs
            onTabChanged={[Function]}
            selectedTab={0}
            tabs={
              Array [
                "Status",
                "Categorie",
              ]
            }
          />,
          "className": "col-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onTabChanged": [Function],
            "selectedTab": 0,
            "tabs": Array [
              "Status",
              "Categorie",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="incident-detail-page__tab-container"
          >
            <withReducer(withSaga(Connect(IncidentStatusContainer))) />
          </div>,
          "className": "col-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <withReducer(withSaga(Connect(IncidentStatusContainer))) />,
            "className": "incident-detail-page__tab-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-12"
          >
            <h3>
              Melding 
            </h3>
          </div>,
          <ul
            className="col-4 incident-detail-page__map"
          >
            <MapDetail
              label=""
            />
          </ul>,
          <div
            className="col-8"
          >
            (
            <Link
              replace={false}
              to="undefined/incidents"
            >
              Terug naar overzicht
            </Link>
            )
            <IncidentDetail
              incident={Object {}}
            />
          </div>,
          <div
            className="col-12"
          >
            <Tabs
              onTabChanged={[Function]}
              selectedTab={0}
              tabs={
                Array [
                  "Status",
                  "Categorie",
                ]
              }
            />
          </div>,
          <div
            className="col-12"
          >
            <div
              className="incident-detail-page__tab-container"
            >
              <withReducer(withSaga(Connect(IncidentStatusContainer))) />
            </div>
          </div>,
        ],
        "className": "incident-detail-page row container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              Melding 
            </h3>,
            "className": "col-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Melding ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Melding ",
              undefined,
            ],
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MapDetail
              label=""
            />,
            "className": "col-4 incident-detail-page__map",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "(",
              <Link
                replace={false}
                to="undefined/incidents"
              >
                Terug naar overzicht
              </Link>,
              ")",
              <IncidentDetail
                incident={Object {}}
              />,
            ],
            "className": "col-8",
          },
          "ref": null,
          "rendered": Array [
            "(",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Terug naar overzicht",
                "replace": false,
                "to": "undefined/incidents",
              },
              "ref": null,
              "rendered": "Terug naar overzicht",
              "type": [Function],
            },
            ")",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "incident": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Tabs
              onTabChanged={[Function]}
              selectedTab={0}
              tabs={
                Array [
                  "Status",
                  "Categorie",
                ]
              }
            />,
            "className": "col-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onTabChanged": [Function],
              "selectedTab": 0,
              "tabs": Array [
                "Status",
                "Categorie",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="incident-detail-page__tab-container"
            >
              <withReducer(withSaga(Connect(IncidentStatusContainer))) />
            </div>,
            "className": "col-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <withReducer(withSaga(Connect(IncidentStatusContainer))) />,
              "className": "incident-detail-page__tab-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
